src/fastfilter.h

```diff
@@ -95,7 +124,8 @@ class CRollingFastFilter : public CFastFilter<FILTER_SIZE>
         erase = insecure_rand.rand32() % CFastFilter<FILTER_SIZE>::FILTER_BYTES;
         eraseAmt = _eraseAmt;
     }
-    void insert(const uint256 &hash)
+
+    void roll(void)
```

Comment:
since the input hashes are unbiased, they are placed randomly within the bloom filter.  therefore a sequential roll is unbiased.  I'm not sure exactly what kind of test you have envisioned but you should look at what I have in the fastfilter_tests.cpp file.

Reply:
I am worried about interaction of `roll()` and an adversary.

Reply:
> That adversary would only penalize his own tx.

Yes, if the fast filter is only used to filter out transactions from a peer. My worry is that its usage grows and implicit assumptions about its hardness are made and at some point we're open to an attack.

> Additionally in looking at the current bloom implementation, I've realized that it's hash is not cryptograpically strong.

As far as I can see, it is not the cryptographically strong that is important here, rather that the hash function is unpredictable from the attacker's POV (e.g. how siphash is described here: https://en.wikipedia.org/wiki/SipHash) 

I don't know whether murmurhash in `CBloomFilter` has that property.

I guess this could all be subsumed under 'be careful when using this' and corresponding warnings in the code.